Tools/Algorithms/stencil.py

```diff
@@ -142,87 +144,115 @@ def compute_stencils(coeff_nodal, coeff_stagg, axis):
+        if axis == 0:
+              # Averaged over ky and kz
+              stencil_avg_nodal  = np.sum(np.sum(stencil_nodal, axis=-1), axis=1)
+              stencil_avg_nodal /= (stencil_nodal.shape[-1]*stencil_nodal.shape[1])
+              stencil_avg_stagg  = np.sum(np.sum(stencil_stagg, axis=-1), axis=1)
+              stencil_avg_stagg /= (stencil_stagg.shape[-1]*stencil_stagg.shape[1])
+        elif axis == 1:
+              # Averaged over kx and kz
+              stencil_avg_nodal  = np.sum(np.sum(stencil_nodal, axis=-1), axis=0)
+              stencil_avg_nodal /= (stencil_nodal.shape[-1]*stencil_nodal.shape[0])
+              stencil_avg_stagg  = np.sum(np.sum(stencil_stagg, axis=-1), axis=0)
+              stencil_avg_stagg /= (stencil_stagg.shape[-1]*stencil_stagg.shape[0])
+        elif axis == 2 or axis == -1:
+              # Averaged over kx and ky
+              stencil_avg_nodal  = np.sum(np.sum(stencil_nodal, axis=1), axis=0)
+              stencil_avg_nodal /= (stencil_nodal.shape[1]*stencil_nodal.shape[0])
+              stencil_avg_stagg  = np.sum(np.sum(stencil_stagg, axis=1), axis=0)
+              stencil_avg_stagg /= (stencil_stagg.shape[1]*stencil_stagg.shape[0])
 
     stencils = dict()
     stencils['nodal'] = abs(stencil_avg_nodal)
     stencils['stagg'] = abs(stencil_avg_stagg)
 
     return stencils
 
-def compute_all(dx, dy, dz, dt, nox, noy, noz, v_gal, nx=256, ny=256, nz=256):
+def compute_all(dims, dx_boosted, dt, psatd_order, v_gal, nx=None):
     """
     Compute nodal and staggered stencils along all directions.
 
     Parameters
     ----------
-    dx : float
-        Cell size along x.
-    dy : float
-        Cell size along y.
-    dz : float
-        Cell size along z.
+    dims : int
+        Number of dimensions.
+    dx_boosted : np.ndarray (float)
+        Cell size along each direction.
     dt : float
         Time step.
-    nox : int
-        Spectral order along x.
-    noy : int
-        Spectral order along y.
-    noz : int
-        Spectral order along z.
+    psatd_order : np.ndarray (int)
+        Spectral order along each direction.
     v_gal : float
         Galilean velocity.
-    nx : int, optional (default = 256)
-        Number of mesh points along x.
-    ny : int, optional (default = 256)
-        Number of mesh points along y.
-    nz : int, optional (default = 256)
-        Number of mesh points along z.
+    nx : np.ndarray (int), optional (default = 256 in each direction)
+        Number of mesh points along each direction.
 
     Returns
     -------
     (stencil_nodal, stencil_stagg) : tuple
         Nodal and staggered stencils along all directions.
     """
+    # Default value for nx
+    if nx is None:
+        nx = np.full(shape=dims, fill_value=256)
+
     # k vectors
-    kx_arr = 2*np.pi*np.fft.fftfreq(nx, dx)
-    ky_arr = 2*np.pi*np.fft.fftfreq(ny, dy)
-    kz_arr = 2*np.pi*np.fft.fftfreq(nz, dz)
+    kx_arr = 2*np.pi*np.fft.fftfreq(nx[0], dx_boosted[0])
```

Comment:
Could these be put in a list, making simpler code like this? Same for below.
```
k_arr = []
for i in dims:
    k_arr.append(2*np.pi*np.fft.fftfreq(nx[i], dx_boosted[I])
```

Reply:
Thank you for the suggestion. I've changed the code to use lists wherever possible, avoiding conditional statements on dimensionality.